apps/web/src/client/components/file-uploader/_ui.js

```diff
@@ -0,0 +1,109 @@
+/**
+ * UI features (file list and choose files button)
+ *
+ * @param {Element} uploadForm
+ */
+const registerUIEvents = (uploadForm) => {
+	/** @type {HTMLElement | null} */
+	const uploadButton = uploadForm.querySelector('.pins-file-upload--button');
+	/** @type {HTMLElement | null} */
+	const uploadInput = uploadForm.querySelector('input[name="files"]');
+	/** @type {HTMLElement | null} */
+	const uploadCounter = uploadForm.querySelector('.pins-file-upload--counter');
+	/** @type {HTMLElement | null} */
+	const filesRows = uploadForm.querySelector('.pins-file-upload--files-rows');
+
+	if (!uploadButton || !uploadInput || !filesRows || !uploadCounter) return;
+
+	const init = () => {
+		uploadButton.addEventListener('click', (clickEvent) => {
+			clickEvent.preventDefault();
+			uploadInput.click();
+		});
+		uploadInput.addEventListener('change', handleFileSelect, false);
+	};
+
+	/**
+	 * Execute actions after selecting the files to upload
+	 *
+	 * @param {*} event
+	 */
+	const handleFileSelect = (event) => {
+		const reader = new FileReader();
+		const { target } = event;
+		const { files } = target;
+
+		reader.addEventListener('load', () => {
+			updateFilesRows(files);
+			updateButtonText();
+		});
+		reader.readAsText(files[0]);
+	};
+
+	/**
+	 * Update button text and files counter
+	 *
+	 */
+	const updateButtonText = () => {
+		setTimeout(() => {
```

Comment:
why the setTimeout?

Reply:
to make sure the counter activates after the files list has finished its changes. could have done it with a Promise but sounded too much effort for such a small feature